src/com/maddyhome/idea/vim/ex/CommandParser.java

```diff
@@ -163,13 +165,24 @@ public boolean processLastCommand(@NotNull Editor editor, @NotNull DataContext c
    * @throws ExException if any part of the command is invalid or unknown
    */
   public int processCommand(@NotNull Editor editor, @NotNull DataContext context, @NotNull String cmd,
-                            int count) throws ExException {
+                            int count, boolean isAlias) throws ExException {
     // Nothing entered
     int result = 0;
     if (cmd.length() == 0) {
       return result | RES_EMPTY;
     }
 
+    // If there is a command alias for the entered text, then process the alias and return that
+    // instead of the original command. We also only want to do this if the previous command
+    // wasn't already an alias. Basically, an alias cannot reference an alias. (Just to be safe)
+    if (VimPlugin.getCommand().hasAlias(cmd)) {
```

Comment:
Weeell... Adding a new boolean to arguments is always a sign of "think a bit more" for me :)
And as I can see, vim allows you to create aliases with existing aliases.

Probably you can use [this](https://stackoverflow.com/a/11300253/3124227) approach?
The first method would be untouched
```
public int processCommand(@NotNull Editor editor, @NotNull DataContext context, @NotNull String cmd,
                            int count) {
    return processCommand(editor, context, cmd, count, 100);
}
```

and the second one with actual logic
```
private int processCommand(@NotNull Editor editor, @NotNull DataContext context, @NotNull String cmd,
                            int count, int aliasCountdown) {
    ...
    if (hasAlias) {
        return processCommand(editor, context, cmd, count, aliasCountdown - 1);
    }
    ...
}
```

Reply:
I like this much more, basically limits the nesting to 100 aliases.

Reply:
I added a limit based on your suggestion. I still need to add tests though to make sure it works.